### PR TITLE
[minor] make backward assert a bit better

### PR DESCRIPTION
**Pull request by min-xu-ai**
## What does this PR do?

Backward assert can be confusing to debug. This helps a bit. Can be more helpful for sure in the future.

@ngoyal2707 @tmarkstrum @anj-s 

## Before submitting

- [ ] Did you have fun?
  - Make sure you had fun coding 🙃
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/main/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [ ] N/A
- [ ] Did you make sure to update the docs?
  - [ ] N/A
- [ ] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/main/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py [hunk 1]
```diff
@@ -1729,7 +1730,8 @@ def _wait_for_post_backward(self) -> None:
         if self._require_backward_grad_sync:
             # Flush any unreduced buckets in the post_backward stream.
             with torch.cuda.stream(self._streams["post_backward"]):
-                assert self._reducer is not None
+                p_assert(self._reducer is not None, "WFPB: reducer is None")
```
Comment (tmarkstrum) [inline fairscale/nn/data_parallel/fully_sharded_data_parallel.py:1733]: What does WFPB mean?

**Review by tmarkstrum — APPROVED** [review]
Other than I am not sure what "WFPB" mean, ILGM.

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py [hunk 2]
```diff
@@ -2355,6 +2364,13 @@ def cpu_offload(self) -> bool:
         return self.move_params_to_cpu
 
 
+def p_assert(cond: Any, s: Any) -> None:
+    """Used in backward context to make sure error is printed."""
```
Comment (anj-s) [inline fairscale/nn/data_parallel/fully_sharded_data_parallel.py:2368]: Doesn't assert already allow you to print an optional message if the condition fails? maybe i am missing something

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py (hunk 1 above)
Comment (min-xu-ai) [reply fairscale/nn/data_parallel/fully_sharded_data_parallel.py:1733]: it is Wait-For-Post-Backward hahaha ;-)

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py (hunk 2 above)
Comment (min-xu-ai) [reply fairscale/nn/data_parallel/fully_sharded_data_parallel.py:2368]: yes, it got swallowed in the backward pass due to the way autograd work.

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py (hunk 1 above)
Comment (anj-s) [reply fairscale/nn/data_parallel/fully_sharded_data_parallel.py:1733]: Hahah I couldn't figure it out either. maybe we should spell it out for future readers.

**Pull request merged**